service/matching/physical_task_queue_key.go

```diff
@@ -31,16 +31,18 @@ import (
 	"strconv"
 	"strings"
 
+	"go.temporal.io/api/deployment/v1"
 	enumspb "go.temporal.io/api/enums/v1"
 	"go.temporal.io/server/common/tqid"
 )
 
 const (
 	// nonRootPartitionPrefix is the prefix for all mangled task queue names.
-	nonRootPartitionPrefix = "/_sys/"
-	partitionDelimiter     = "/"
-	versionSetDelimiter    = ":"
-	buildIdDelimiter       = "#"
+	nonRootPartitionPrefix  = "/_sys/"
+	partitionDelimiter      = "/"
+	versionSetDelimiter     = ":"
+	buildIdDelimiter        = "#"
+	deploymentNameDelimiter = "|"
```

Comment:
This is really cool! Let's aim to draw something meaningful here. Like a Christmas tree.